docs/shared/recipes/tips-n-tricks/switch-to-workspaces-project-references.md

```diff
@@ -172,26 +172,79 @@ The root `tsconfig.json` file should extend `tsconfig.base.json` and not include
 {% /tab %}
 {% /tabs %}
 
+## Register Nx Typescript Plugin
+
+Make sure that the `@nx/js` plugin is installed in your repository and `@nx/js/typescript` is registered as a plugin in the `nx.json` file.
+
+```jsonc {% fileName="nx.json" %}
+{
+  "plugins": [
+    {
+      "plugin": "@nx/js/typescript",
+      "options": {
+        "typecheck": {
+          "targetName": "typecheck"
+        },
+        "build": {
+          "targetName": "build",
+          "configName": "tsconfig.lib.json",
+          "buildDepsName": "build-deps",
+          "watchDepsName": "watch-deps"
+        }
+      }
+    }
+  ]
+}
+```
+
+This plugin will register a [sync generator](/concepts/sync-generators) to automatically maintain the project references across the repository.
+
 ## Create Individual Project package.json files
 
 When using package manager project linking, every project needs to have a `package.json` file. You can leave all the task configuration in the existing `project.json` file. For application projects, you only need to specify the `name` property. For library projects, you should add an `exports` property that accounts for any TypeScript path aliases that referenced the project. A typical configuration is shown below:
 
+{% tabs %}
+{% tab label="Non-buildable library" %}
+
+```json {% fileName="libs/ui/package.json" %}
+{
+  "name": "@myorg/ui",
+  "exports": {
+    ".": {
+      "types": "./src/index.d.ts",
+      "import": "./src/index.js",
```

Comment:
I would suggest to add a short sentence explaining the difference of when to use *.js and when to use *.ts here.
(As far as I understood: Use *.ts if the lib is not buildable, use *.js if it is buildable?)

Reply:
actually just saw right now that these new sections are non-buildable vs. buildable, so please double-check if these settings are correct.

Reply:
Good catch.  I had the wrong settings for non-buildable